test/functional/placeholder.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2015 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var IncrementalDOM = require('../../index'),
+    patch = IncrementalDOM.patch,
+    elementOpen = IncrementalDOM.elementOpen,
+    elementClose = IncrementalDOM.elementClose,
+    elementVoid = IncrementalDOM.elementVoid,
+    elementPlaceholder = IncrementalDOM.elementPlaceholder;
+
+describe('placeholders', () => {
+  var container;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  describe('without an existing DOM element', () => {
+    var el;
+
+    beforeEach(() => {
+      patch(container, () => {
+        el = elementPlaceholder('div', 'placeholder', ['class', 'someClass'], 'data-foo', 'bar');
+      });
+    });
+
+    it('should render with the specified tag', () => {
+      expect(el.tagName).to.equal('DIV');
+    });
+
+    it('should render with static attributes', () => {
+      expect(el.className).to.equal('someClass');
+    });
+
+    it('should not render with dynamic attributes', () => {
+      expect(el.getAttribute('data-foo')).not.to.equal('bar');
+    });
+  });
+
+  describe('with an existing DOM element', () => {
+    var div;
+    var el;
+    var child;
+
+    beforeEach(() => {
+      patch(container, function() {
+        div = elementPlaceholder('div', 'placeholder');
+        patch(div, function() {
+          child = elementVoid('div', null, ['id', 'child']);
+        });
+      });
+
+      patch(container, () => {
+        el = elementPlaceholder('span', 'placeholder', ['class', 'someClass'], 'data-foo', 'bar');
```

Comment:
> Would it be possible create a wrapping function that creates it with the correct tag and display: inline instead? Or maybe just manually pass it.

Subview's aren't known to the parent until one is inserted and, even then, another subview can replace it.

```html
<!-- Tab 1's template, including root -->
<ul>...</ul>
<!-- tab 1 -->

<!-- Tab 2's template, including root -->
<section><p>...</p></section>
<!-- tab 2 -->


<div id="parent">
  <nav><a href="#tab1">Tab 1</a> | <a href="#tab2">Tab 2</a></nav>
  <span key="placeholder"></span>
</div>
```

Wrapping divs would be the only generic solution here, but personally despise them.

> You would expect that if the condition changes that the tag should change as well.

For an `elementVoid`, yes. But I'm trying to define placeholders as generic identifiers that _just_ hold a position in the DOM tree, since it's impossible to track the position without inserting some element in that position. What's currently occupying that position shouldn't matter, as long as the position is maintained.